### PR TITLE
[SPMD] Use xs.Mesh in test_2d_tensor_3d_mesh

### PR DESCRIPTION
**Pull request by khatwanimohit**
Use `Mesh` in the test for 2d_tensor_3d_mesh until V2 support is added in `HybridMesh`

### CODE REVIEW

File: test/spmd/test_xla_sharding.py [hunk 1]
```diff
@@ -481,7 +481,6 @@ def test_xla_sharded_hlo_dump(self):
     # scalar 5 should be replicated
     self.assertIn('%p0.2 = f32[] parameter(0), sharding={replicated}', hlo)
 
-  @unittest.skipIf(xr.device_type() == 'TPU', "Crash on TPU v2")
   @patch('torch_xla.runtime.global_device_attributes')
   @patch('torch_xla.core.xla_model.xla_device_hw')
```
Comment (jonb377) [inline test/spmd/test_xla_sharding.py:485]: Since we're not using HybridMesh, we can remove the mocking logic here

**Review by jonb377 — APPROVED** [review]
Thanks Mohit!

File: test/spmd/test_xla_sharding.py [hunk 2]
```diff
@@ -505,8 +504,7 @@ def test_2d_tensor_3d_mesh(self, xla_device_mock, device_attributes_mock):
 
     t1 = ct1.to(xm.xla_device())
     t2 = ct2.to(xm.xla_device())
-    mesh = self._get_hybrid_mesh((1, self.n_devices, 1),
-                                 axis_names=('data', 'fsdp', 'tensor'))
+    mesh = self._get_mesh((1, self.n_devices, 1))
```
[the remark this answers is not in the document]
Comment (alanwaketan) [reply test/spmd/test_xla_sharding.py:491]: Saw that it's covered by `test_hybrid_mesh_shape`.

**Review by alanwaketan — APPROVED** [review]

**Comment by JackCaoG** [comment]
waiting for TPU CI before merge.

**Pull request merged**